File: javascript/044_secret-handshake/secret-handshake.js
export const commands = (number) => {
  let binary = number.toString(2).padStart(5,0).split('').reverse().map(Number);
  const secret = ['wink','double blink','close your eyes','jump'].filter((_,i) => binary[i]);
  return binary.at(-1) ? secret.reverse() : secret;
};

File: python/ex_038_secret-handshake/secret_handshake.py
def commands(binary_str):
    code = ('jump', 'close your eyes', 'double blink', 'wink')
    secret = []
    for index, bit in enumerate(binary_str[-4:]):
        if int(bit):
            if int(binary_str[-5]):
                secret.append(code[index])
            else:
                secret.insert(0,code[index])      
    return secret